crates/kitsune_p2p/types/src/metrics.rs

```diff
@@ -0,0 +1,175 @@
+//! Utilities for helping with metric tracking.
+
+use std::sync::{
+    atomic::{AtomicU64, AtomicUsize, Ordering},
+    Once,
+};
+use sysinfo::{NetworkExt, NetworksExt, ProcessExt, SystemExt};
+
+static SYS_INFO: Once = Once::new();
+
+static TASK_COUNT: AtomicUsize = AtomicUsize::new(0);
+static USED_MEM_KB: AtomicU64 = AtomicU64::new(0);
+static PROC_CPU_USAGE_PCT_1000: AtomicUsize = AtomicUsize::new(0);
+static TX_BYTES_PER_SEC: AtomicU64 = AtomicU64::new(0);
+static RX_BYTES_PER_SEC: AtomicU64 = AtomicU64::new(0);
+
+/// Spawns a tokio task with given future/async block.
+/// Captures a new TaskCounter instance to track task count.
+#[macro_export]
+macro_rules! metric_task {
+    ($task:expr) => {{
+        let counter = $crate::metrics::MetricTaskCounter::new();
+        let fut = { $task };
+        ::tokio::task::spawn(async move {
```

Comment:
```suggestion
        $crate::dependencies::tokio::task::spawn(async move {
```

we don't know that they directly imported tokio...